core/controllers/collection_editor_test.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the collection editor page."""
+
+__author__ = 'Abraham Mgowano'
+
+
+import os
+import StringIO
+import zipfile
+
+from core.controllers import collection_editor
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.domain import config_services
+from core.domain import rights_manager
+from core.domain import rule_domain
+from core.tests import test_utils
+import feconf
+import logging
+
+
+class BaseCollectionEditorControllerTest(test_utils.GenericTestBase):
+
+    CAN_EDIT_STR = 'GLOBALS.can_edit = JSON.parse(\'true\');'
+    CANNOT_EDIT_STR = 'GLOBALS.can_edit = JSON.parse(\'false\');'
+
+    def setUp(self):
+        """Completes the sign-up process for self.EDITOR_EMAIL."""
+        super(BaseCollectionEditorControllerTest, self).setUp()
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+
+        self.set_admins([self.ADMIN_EMAIL])
+
+    def assert_can_edit(self, response_body):
+        """Returns True if the response body indicates that the collection is
+        editable.
+        """
+        self.assertIn(self.CAN_EDIT_STR, response_body)
+        self.assertNotIn(self.CANNOT_EDIT_STR, response_body)
+
+    def assert_cannot_edit(self, response_body):
+        """Returns True if the response body indicates that the collection is
+        not editable.
+        """
+        self.assertIn(self.CANNOT_EDIT_STR, response_body)
+        self.assertNotIn(self.CAN_EDIT_STR, response_body)
+
+
+class CollectionEditorTest(BaseCollectionEditorControllerTest):
+    def setUp(self):
+        super(CollectionEditorTest, self).setUp()
+
+        self.COLLECTION_ID = '0'
+
+        collection_services.load_demo(self.COLLECTION_ID)
+        rights_manager.release_ownership_of_collection(
+            feconf.SYSTEM_COMMITTER_ID, self.COLLECTION_ID)
+
+    def test_access_collection_editor_page(self):
+        """Test access to editor pages for the sample collection."""
+
+
```

Comment:
Nit: Remove newline (there should be just 1 between the docstring and the ``# Check that it...`` comment).